### DISCUSSION TITLE
Node balancing

### DISCUSSION THREAD

**Discussion by brian-man** [sunflower-land/sunflower-land#2263] [General]
Blasta raised this issue in a thread under `the-sunflorum` on Discord but it hasn't gotten really any dev attention from what I can tell.  I'm reopening here (with Blasta's blessing) to hopefully get some traction on this problem area.  I didn't capture the whole discussion here, just the opening post.

You can find the original thread here:
https://discord.com/channels/880987707214544966/1063925419738407012

### Problem statement [Blasta]
 — 01/14/2023 12:59 PM

The recent update allowing for the 12th land expansion to add a bit of balance to the RNG in regards to Fruit Trees, Plots and regular trees… I believe, was an awesome call (coming from someone that usually has very bad luck on RNG). Obviously there was a bit of imbalance between fully expanded farms with 31 plots vs those with 44.

Personally I was attracted to Sunflower Land at the beginning because there was more skill/strategy based gameplay, and very little RNG, which lacks in a lot of games in the Web3 environment. The recent addition of all the RNG elements has been a bit of a struggle to adjust too for me.

All that aside the purpose of this post is to ask/explore the idea of a balancing patch for the stone/iron and gold nodes. I believe a greater imbalance exists with these between fully expanded farms, than that was with the plots as some have literally more than double the amount of nodes than others.

For example, some have 4 irons nodes and 3 gold (fully expanded), but others have 10 irons and 6 gold….
With all the new NFTs requiring Gold, this imbalance is make the playing field pretty unfair.

Is there plans in the pipeline to try and balance this out? And does anyone think this is needed like it was with the Fruit and plots?

**Comment by brian-man** [comment]
— 02/03/2023 11:16 AM
In theory...
1. It should be balanced such that the 10 gold node guy has crap for plots (or patches or whatever).
2. The team should have a model for the relative value of node types and the budget should be not-necessarily-linear for node type X vs. Y.  Meaning, for example, maybe Gold nodes use 5 points of budget and Iron nodes use 3 points.  I really expected this to be the design when Steve was talking about planning it all out months ago.
https://discord.com/channels/880987707214544966/880987707692687381/1070176826070159412

**Comment by brian-man** [comment]
— 02/03/2023 11:26 AM
If they decided to move to a model like I described, I think it would be a fairly simple process to "fix" existing islands by doing something like:
3. On March 1st, tally up the value of each land.  The highest point value for each expansion level sets the standard.  For all farms of expansion level X, credit points via coupons to the land owner to "catch the land" up to the standard.  Introduce UI to allow spending the coupons (with the hammer?) to install nodes according to the budgeted allotment.

And optionally:
4. Introduce "discard node" feature that allows you to remove a node and be credited the appropriate coupons.  Players could then use the same UI in 3 to organize their land nodes to their playstyles.

**Comment by brian-man** [comment]
Dcol:

— Yesterday at 4:34 AM
The current rolls according to Inu's sheet are:

Good - 10%
Average - 20%
'Bad' - 70%

I think it should be:

Good - 12.5%
Average - 75%
'Bad' - 12.5%

**Comment by brian-man** [comment]
— Yesterday at 6:59 AM
> Good - 10%
> Average - 20%
> 'Bad' - 70%

So with each expansion 70% of the player base is motivated to leave.  That seems like a bad design (no pun intended).
﻿
The RNG should affect *variety* not *quality* of an expansion, IMO.
﻿
Example of variety:
A: I got 1 gold.
B: I got 2 stone and 1 iron.
C: I got 8 plots.
﻿
Example of quality:
A: I got 1 gold, 1 iron, 1 stone, and 2 plots.
B: I got 1 stone.

### Proposal
﻿
My personal preference is a point system model.  Perhaps you get a starting node assignment with the points and can pay (shovel or whatever) to convert the nodes to points for spending on new nodes.
﻿
As far as implementation, I would probably simplify to:
1. for each expansion available, assign an intended point value
2. for each farm, calculate the used points for each expansion
3. comparing 2 with 1, credit each farm with the deficit in points (perhaps disabling minting new expansions for a couple hours while tallying and applying the points)
4. going forward either:
   a. assign points when expansions happen (rather than nodes)
   OR
   b. randomly assign acc to 1, and provide UI for converting nodes to points and vice-versa
﻿

I've described something like this at least 4 times. I don't think this is complicated and I think it pretty much addresses every complaint I've seen on discord regarding expansion RNG.

**Comment by Aminbtdds** [comment]
I saw your proposal mechanic at least 3 times at discord, and it seems logical in terms of balancing ore nodes, it is not right that some lands overall performance be 200% or 300% more than others in same expansion level.
It is not variation, it is unfairness and bad game design.
In the last expansion people with 3 iron node got zero nodes, but some players with 5 iron nodes got another and became 6 nodes, looks like instead of minimizing disparity and node count spread, our current RNG system keep widening the spread, which will result in massive "losers regret" mindset of players after more expansions, which will eventually result in reducing game retention or changing player type to extractors or encouraging multiaccounts or account flippers as a result.
Beside your point system which i like, if devs feel they have less control on overall game node count, game design at new expansions can check current node count of farms and reroll at mint IF the new piece makes spread wider, for example at last piece of land:
If Options of new land was👇
A. 2 plot 1 fruit
B. 1 iron 1 fruit
C.1 plot 1 iron 1 fruit

At the roll for farm👇

X: 10 tree 4iron 3gold

If The median of iron nodes is 4, and farm X got B for its roll, it will result to more than median nodes which is 4 and the farm already have 4 iron, so the system reroll to get A or C.

Y: 12 tree 3iron 4gold

The median of iron is not met, which makes farm Y eligible for roll B or C and not A.
.
.
.

We would still have diversity but with less disparity and players feeling "loser regret".
Also it will encourage players more to expand their lands to balance their overall node count by tending to median node count at same levels.
P.s.
Imho, we are going far away from what devs told about game being "escape from reality" kind of game👌 nobody escape from reality to feel like losers repeatedly. Also we were one crypto game which didn't look like Casino, with having RNGs in every new aspect of gameplay, sfl will look like casino which makes us gamblers🤷‍♂️ lots of players i know love sfl because they feel strategizing not gambling🌻❤🙏

**Reply by Mehrasa13** [reply]
Although I have 7 iron and 4 gold.  I agree with this proposal. It seems like a reasonable and fair proposal🌻

**Reply by Mehran7583** [reply]
Excellent. I agree with your proposal too.👌👏

**Reply by sabryder** [reply]
I have 6 iron, 3 gold
overall it is a good and decent amount but in comparison to the one with highest i have nothing😅😂
And again in comparison to the one with lowest,I have double the amount of iron node.
I dont disagree with luck based expansions but this rng system is out of control, there should be luck based and balance way to do this
so amin, i completely agree with you as always❤️❤️👌🏻

**Reply by Rook982** [reply]
Up for this

**Reply by Ogrekin** [reply]
+1 up vote!

**Reply by Droidze** [reply]
I agree with your idea Amin, new players who come to Sunflower Lands after spending some time expanding their lands and getting unfair rolls for the number of nodes in each land feel disappointed and think they are gambling on their money and time.

  Currently, it may not be so important from the team's point of view to include this one in the todo proposals, but ignoring it can cause irreparable damage to the entire project shortly, so fair improvements should be made in this case as soon as possible.

  As a solution, when players mint new lands, instead of randomly receiving lands, they can be given two or three fair choices to roll based on the average of data collected from other farms' node count spread, so they can choose one according to their interests, nfts they own, or other things. these lands are very close to the Median distribution of available farms node and can contain a few more nodes, plots, or fruits, so players have better control over deciding which path they want to go. ❤️❤️

**Reply by ShiningMinions** [reply]
@Aminbtdds  I love your post :heart: , but I do disagree with some of it.

> ...it is not right that some lands overall performance be 200% or 300% more than others in same expansion level.

I agree with this when everyone is stuck with whatever lands they get. I disagree with this whenever players have some control over their land nodes.

> It is not variation, it is unfairness and bad game design.

Let's be clear, it was not "unfair". It may feel this way, but the truth is that it was very fair - you were just unlucky. You had equally as much of a chance to roll a good land as everyone else. It did not matter how rich your were, what your ethnicity is, your gender, race, etc. IMO, the reason it FEELS unfair, is because you are forever stuck with whatever lands you got. This is what needs to change.

> In the last expansion people with 3 iron node got zero nodes, but some players with 5 iron nodes got another and became 6 nodes, looks like instead of minimizing disparity and node count spread, our current RNG system keep widening the spread, which will result in massive "losers regret" mindset of players after more expansions, which will eventually result in reducing game retention or changing player type to extractors or encouraging multiaccounts or account flippers as a result.

> ...nobody escape from reality to feel like losers repeatedly.

Again, the problem is NOT "RNG widening the disparity". The real problem (and the reason you feel like a loser repeatedly) is that you are forever stuck with whatever expansions you get. You're not allowed to start a new farm (multi-account ban). You're not able to trade landTiles for other tiles you actually want. You have 0 options, leading to bitter and envious feelings towards others who actually did get lucky.

**Comment by Ogrekin** [comment]
I agree with this!
With current system there have become a to big difference between farms and to big for it to be accepted as an 
"unlucky draw, no big deal"..
Now it has almost gotten to that you don't wan't to mention how many nodes you have becauce you feel bad for those that got unlucky..
There comes more and more nft that requires gold and some people will have a lot harder to keep up...
Old players can probably accept to an bigger extent but for a new player that joins and ends up with lowest gold nodes to see others 
with 200-300% more gold node would for sure feel screwed.. 

How it should be done that have them that are good at things like this figuare out ( I'm not good at it ) but i really see the need that this has to be adressed as soon as possible before we go to far and have many new players with expanded farms to feel what many already
do... We have to many as it is now that want to change farm and do so to...

If this isn't handled soon we probably could have a big problem with flipping farms, multi accounting and a lot of unhappy players that
discord members have to deal with instead of take care of new that comes... It is already much talk about it on discord and it has become
an more negative attitude all together so i hope this will have high priority and will follow it and looking forward to read peoples ideas how the balancing should be,, we have many that are extremely smart and bright in discord and have good ideas..

**Comment by Rook982** [comment]
I agree with this, and I have an above average farm (43 plots, 12 trees, 9 stone, 6 irons, 5 gold, latest expansion).

I know the plan for Land Expansions was for farms to have that feeling of being "rare" that is common with the NFT space. But with the current system it will become targeted by flippers and extractors instead of people who steadily play the game. Catch-up mechanics are great but having a level playing field for everyone introduces more strategy and planning for the fierce competition for the NFTs. Why I loved the fruit patch balancing land. It gave me a shot to get the Black Bearry and I did, but its because I adjusted my life around the Blueberry schedule not because I was so ahead that I was a sure mint.

Please look into the posts made by those awesome people above too that involve having a Resource Point System for Lands. My additional suggestion would be where your land is influenced by your skills. Say I have most of my points in Farming then there would be a modifier that increases my farm plots in the next expansion, same can apply with Trees and Stones. This brings into another Bumpkin feature where having multiple bumpkins affect your overall farm in the long run.

**Reply by Rook982** [reply]
https://github.com/sunflower-land/sunflower-land/discussions/2180#discussioncomment-5016344

On a pinned post where they discuss new skill trees, I commented where they could also add skills that just straight up add resource nodes as a skill. Further reinforces specialization and reduces RNG reliance.

**Comment by MrPunish33r** [comment]
I wonder if it's really hard to just add another expansion to add some balance to the nodes, just like they did with the plots. for ex people who have 3 gold nodes will get a +1 or +2.... we're really struggling with resources, especially gold as it's the hardest to mine and takes longer to reset.

**Comment by Xeko-92** [comment]
The only issue I see with this is things like foreman beaver. If people get complete freedom to choose what they want with their nodes, people with foreman beaver will just go all trees. So there needs to be some sort of limitations of how many of each type of resource people can have.

**Reply by Rook982** [reply]
They should be free to do so I think as FB is a high value NFT, with hoarding limits still active, they would still need to sync every x amount of trees they gain no matter how many trees they have. They would have to sync more often now too since they have so many trees.

Best way to build around it is to introduce more sinks for resources. Thats my take on it as people who own high value NFTs need to be compensated for their time and effort.

**Reply by Aminbtdds** [reply]
As a FB owner i prefer my farm having node counts ratio be in balance instead of becoming wood provider for market👌😄
Gold production cost is 0.6 sfl in my farm, pretty good margin for balanced farm with endpoint product of gold.
Not all FB owners use it 16hrs a day, i usually chop 8 to 10 times a day.
FB also have a low supply, imo, even 300 specialized only tree farms can't hurt wood supply balance.

**Comment by TheDcol** [comment]
The current system is flawed, otherwise we would not have received an equalizer expansion at the 12th land.

I think paying for rerolls is also not the best solution, as I believe two people should have very similar opportunities.

Take this example, Tony and Peter:

- Tony expands 10 times and ends up with 10 trees, 10 stones, 5 iron, 5 gold and 5 fruit trees. His total expansion cost was 200sfl

- Peter also expands 10 times but ends up with 7 trees, 7 stones, 3 iron, 3 gold and 3 fruit trees. His total expansion cost was also 200sfl.

Now let's say fruit nfts come out and the race begins.. Tony has an innate advantage due to having 2 more fruit patches than Peter. They both paid the same amount of sfl yet Peter is at a severe disadvantage due to no fault of his own. Is this fair? Is Peter right in being annoyed? (Btw this is why equalizer land was brought out.. so everyone had same fruit patch amounts)

Now, let's say rerolls are introduced for 50sfl each.... Peter spends 2 x rerolls and he manages to get the ALMOST the same amount of nodes as Tony.

Peter and Tony have very similar lands now, but Peter had to spend an extra 100sfl to do achieve this. Is this fair? Is Peter right in being annoyed having to spend more SFL to get to a similar point as Tony?

Now imagine when lands are expanded 100 times... The DIFFERENCE between lands will get even wider and more 'unfair'.

--
I repeatedly stated that changing the chances of the land quality rolls needs to be looked at, so that the variance of lands is not as drastic as now. Highest probability land roll needs to be medium quality, not low quality as it is now. The more people expand the more you will end up with this group of 300-500 players who will have a clear advantage over the rest in nft races, this was evident in the fruit nft race but the devs foresaw this and equalized the lands.

^equalizing land was a good idea, but you would not need to do such manual processing if the initial land roll chances were more balanced to be 70% medium quality, 15% good and 15% bad. You would still end up with unhappy people, no matter what you did, but it would certainly be A LOT LESS annoyed people doing it this way, i believe.

We can also discuss other options such as a points based system someone brought up or trading (although, multis would have field day, maybe level lock? But bumpkins can move around... Ahhhh)

Using points system, 1 plot = 1 points, 1 tree = 5 points...

Another possible solution:
If everyone was assigned same points at expansions with different resources, and then you could trade those nodes with an npc at stonehaven island dor example with diminishing returns?

Meaning, you expand a land and you get 2 trees no plots.. but you dont care about trees.. so you go to the npc and trade 2 trees = 10 points MINUS TAX of 20% so you get 8 points back... and trade for 8 plots. This way, nodes themselves are burnt as you dont get 100% of the return back. A good way to introduce some sort of burn ingame.

You end up with people specialising too.

If everyone was to go for plots... There would be no trees and so those who stuck to wood chopping would need to provide wood for everyone else.
There could be interesting metas/markets around this

Have a good day everyone

**Reply by Aminbtdds** [reply]
Great disscusion👌💐❤🙏
The tax part about point system tourist proposal looks good to prevent players from going back and forth.

**Reply by ShiningMinions** [reply]
> We can also discuss other options such as a points based system someone brought up or trading (although, multis would have field day, maybe level lock? But bumpkins can move around... Ahhhh)

I'm wondering why we can't just have a system for bumpkins similar to how it is for placeables? If you want bumpkins to be used on your farm, then bumpkins should be sent to your farm and placed, just like every other farm NFT. In this way coreDevs have the power to impose cooldowns and other limits so bumpkins can't be abused.

**Reply by TheDcol** [reply]
Would be heavily for this

**Comment by ShiningMinions** [comment]
Unpopular opinion time :smile: 

> All that aside the purpose of this post is to ask/explore the idea of a balancing patch for the stone/iron and gold nodes. I believe a greater imbalance exists with these between fully expanded farms, than that was with the plots as some have literally more than double the amount of nodes than others.

I don't think the main problem is with "balancing". I think the main problem is that whatever land-tiles I receive, I'm forever stuck with - period! I'm not allowed to create a new farm and try again (multi-account ban). I don't have the ability to trade my land tiles for ones I would actually want. I don't have the ability to re-roll my land. I am forever dead-locked in this system, which then makes me bitter and envious of other who actually did get lucky. It is, quite literally, impossible for me to get the farm I want.

> For example, some have 4 irons nodes and 3 gold (fully expanded), but others have 10 irons and 6 gold….
With all the new NFTs requiring Gold, this imbalance is make the playing field pretty unfair.

I honestly don't mind when people get better lands than me. In fact, I am happy that they did! I think this is so awesome and makes the game actually exciting.

But let's be clear: this person getting better lands than you was not "unfair". The truth is you had an equal chance to get these same "good expansions" as they did. The "unfair" part, is what I said in my previous statement: You have 0 options when it comes to getting the land you actually want.
